lib/gems/pending/util/miq_object_storage.rb

```diff
@@ -15,4 +19,33 @@ def initialize(settings)
   def logger
     @logger ||= $log.nil? ? :: Logger.new(STDOUT) : $log
   end
+
+  private
+
+  DONE_READING = ""
+  def read_single_chunk(chunksize = DEFAULT_CHUNKSIZE)
```

Comment:
These methods have some tests attached to them:

https://github.com/ManageIQ/manageiq-gems-pending/compare/master...NickLaMuro:swift-storage-as-object-storage#diff-0f23e1893b4e97a4b0b3e88e288b7c65

Can we get those pulled into here as well?  There are a few other files that you will have to pull in as well as support.

Reply:
For the record, I was planning on breaking this bit out for a separate PR, but since we already have pulled it in to this one, I don't know that it makes sense for the effort to do that at this point.

A better explanation of what is going on here and where this came from can be found in this commit message:  https://github.com/ManageIQ/manageiq-gems-pending/commit/32ee92d473aab0ef5a27fba98b7ae503f3170726

And I probably should have had some inline documentation for some of these methods... sorry about that...

Reply:
@NickLaMuro I cannot find those tests from that link - when I click through it finds no diffs. I'm not sure if I'm supposed to fill in an appropriate branch or head to find them....

Reply:
Sorry, looks like GitHub's markdown syntax had it's link converter mess with that link, here it is in "copy-pasta" form:

```
https://github.com/ManageIQ/manageiq-gems-pending/compare/master...NickLaMuro:swift-storage-as-object-storage#diff-0f23e1893b4e97a4b0b3e88e288b7c65
```